### PR TITLE
Staging release quick cheat sheet.

### PR DESCRIPTION
**Pull request by MihaiTabara**

### CODE REVIEW

File: how-tos/staging-release-setup.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+4. The script should have created a diff. Check that everything is okay and push to jamun
+```sh
+hg -R build/jamun diff
+hg out -r . jamun
+hg push
+```
+
+## Staging tools
+
+- staging [Ship-it](https://ship-it-dev.allizom.org/)
```
Comment (JohanLorenzo) [inline how-tos/staging-release-setup.md]: Nit: dev ship-it. There's a real stage instance at https://ship-it.allizom.org/

File: how-tos/staging-release-setup.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- staging [release runner](https://hg.mozilla.org/build/puppet/file/default/manifests/moco-nodes.pp#l633)
+- release automation notificats [group](https://groups.google.com/a/mozilla.com/forum/?hl=en#!forum/release-automation-notifications-dev) and `#release-notifications-dev` IRC channel
+- staging [balrog](https://balrog-admin.stage.mozaws.net/)
+
+A note on Balrog. Historically, we've had issues with plugging the staging instance to
+the staging release pipeline, so we used another workaround:
+- from aws console, create a new ubuntu AWS instance
+- login to it and clone Balrog [codebase](https://github.com/mozilla/balrog)
+- follow the [installation](https://github.com/mozilla/balrog#installation) process
+- point that instance in all configs [mozharnss configs](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/) that need it. Sample [here](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/dev_updates_firefox_beta.py#23)
```
Comment (JohanLorenzo) [inline how-tos/staging-release-setup.md]: Nit: Use permalinks like https://dxr.mozilla.org/mozilla-central/rev/7d2e89fb92331d7e4296391213c1e63db628e046/testing/mozharness/configs/releases/dev_updates_firefox_beta.py#23

File: how-tos/staging-release-setup.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## Staging configs
+
+- release runner will consume [project branch configs](https://dxr.mozilla.org/build-central/source/buildbot-configs/mozilla/project_branches.py#114)
```
Comment (JohanLorenzo) [inline how-tos/staging-release-setup.md]: Nit: permalink

File: how-tos/staging-release-setup.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- mozharness [configs](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/) - all configs with `dev` prefix
```
Comment (JohanLorenzo) [inline how-tos/staging-release-setup.md]: Nit: permalink

File: how-tos/staging-release-setup.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+```sh
+hg -R build/jamun diff
+hg out -r . jamun
+hg push
+```
+
+## Staging tools
+
+- staging [Ship-it](https://ship-it-dev.allizom.org/)
+- staging [release runner](https://hg.mozilla.org/build/puppet/file/default/manifests/moco-nodes.pp#l633)
```
Comment (JohanLorenzo) [inline how-tos/staging-release-setup.md]: Nit: Use permalinks like: https://dxr.mozilla.org/build-central/rev/5f83e0516fc449586bbce4db4eb759f6cede8781/puppet/manifests/moco-nodes.pp#633

File: how-tos/staging-release-setup.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+- from aws console, create a new ubuntu AWS instance
+- login to it and clone Balrog [codebase](https://github.com/mozilla/balrog)
+- follow the [installation](https://github.com/mozilla/balrog#installation) process
+- point that instance in all configs [mozharnss configs](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/) that need it. Sample [here](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/dev_updates_firefox_beta.py#23)
+
+## Staging configs
+
+- release runner will consume [project branch configs](https://dxr.mozilla.org/build-central/source/buildbot-configs/mozilla/project_branches.py#114)
+- mozharness [configs](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/) - all configs with `dev` prefix
+- in order to avoid pushing patcher configs changes/tags to *real* [tools](http://hg.mozilla.org/build/tools/) repo, please point to a user fork. You can reuse rail's [tools repo](https://hg.mozilla.org/users/raliiev_mozilla.com/tools)
```
Comment (JohanLorenzo) [inline how-tos/staging-release-setup.md]: Nit: rail's link is broken.

After you left, rail showed me how to quickly create a custom repo.
```sh
ssh hg.mozilla.org clone tools-fake build/tools
# Will show: Please wait.  Cloning /build/tools to /users/jlorenzo_mozilla.com/tools-fake
```
I guess it's just better to reclone a repo from there. That'll avoid missing permissions.


Source: https://wiki.mozilla.org/Release:Release_Automation_on_Mercurial:Staging_Specific_Notes#Setup_staging_repos

**Review by JohanLorenzo — APPROVED** [review]
Thank you very much for this big piece of documentation @MihaiTabara ! 😃 

It overall looks good to me. I noticed a few links that might bitrot in the future. We should probably change them to permalinks.

File: how-tos/staging-release-setup.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+3. Run the actual script to get the "diff" between the two branches:
+```sh
+python mozharness-central/scripts/merge_day/gecko_migration.py -c merge_day/central_to_beta.py
+```
+
+4. The script should have created a diff. Check that everything is okay and push to jamun
+```sh
+hg -R build/jamun diff
```
Comment (rail) [inline how-tos/staging-release-setup.md:56]: You need to commit after reviewing the diff:
``` sh
hg -R build/jamun commit -m "Uplift from central to jamun"
```

File: how-tos/staging-release-setup.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+- staging [Ship-it](https://ship-it-dev.allizom.org/)
+- staging [release runner](https://hg.mozilla.org/build/puppet/file/default/manifests/moco-nodes.pp#l633)
+- release automation notificats [group](https://groups.google.com/a/mozilla.com/forum/?hl=en#!forum/release-automation-notifications-dev) and `#release-notifications-dev` IRC channel
+- staging [balrog](https://balrog-admin.stage.mozaws.net/)
+
+A note on Balrog. Historically, we've had issues with plugging the staging instance to
+the staging release pipeline, so we used another workaround:
+- from aws console, create a new ubuntu AWS instance
+- login to it and clone Balrog [codebase](https://github.com/mozilla/balrog)
+- follow the [installation](https://github.com/mozilla/balrog#installation) process
```
Comment (rail) [inline how-tos/staging-release-setup.md]: Before starting the balrog process, we need to set `STAGING=1` to make balrog accept `http://ftp.stage.mozaws.net` in the blobs, something like https://gist.github.com/rail/8488e888273cb1d4af62d469720909c6

File: how-tos/staging-release-setup.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+## Staging configs
+
+- release runner will consume [project branch configs](https://dxr.mozilla.org/build-central/source/buildbot-configs/mozilla/project_branches.py#114)
+- mozharness [configs](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/) - all configs with `dev` prefix
+- in order to avoid pushing patcher configs changes/tags to *real* [tools](http://hg.mozilla.org/build/tools/) repo, please point to a user fork. You can reuse rail's [tools repo](https://hg.mozilla.org/users/raliiev_mozilla.com/tools)
+
+## Misc
+
+TODO - Release runner is smart. It only takes into account the Tier1 stuff.
```
Comment (rail) [inline how-tos/staging-release-setup.md]: Can you also mention that the update verify tests are flaky and we need to invest some time to make them work (I believe by tweaking some variables in https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/dev_updates_firefox_beta.py)

**Review by rail — CHANGES_REQUESTED** [review]
In overall lgtm! Thank you!

File: how-tos/staging-release-setup.md (hunk 4 above)
Comment (MihaiTabara) [reply how-tos/staging-release-setup.md]: Not sure it works on folders. I get Service Temporarily Inactive when I try to do this.

**Comment by MihaiTabara** [comment]
@JohanLorenzo @rail 
Thanks both for detailed reviews and picking all the missing bits. Reiterated with another version to address all your comments. Hopefully I didn't miss anything.

**Review by rail — APPROVED** [review]
👍

**Pull request merged**